dali/kernels/imgproc/paste/paste_gpu.h

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// THIS WHOLE CODE IS A COPY-PASTE OF multiply-add kernel
+
+#ifndef DALI_KERNELS_IMGPROC_PASTE_PASTE_GPU_H_
+#define DALI_KERNELS_IMGPROC_PASTE_PASTE_GPU_H_
+
+#include <vector>
+#include "dali/core/span.h"
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/kernels/imgproc/roi.h"
+#include "dali/kernels/common/block_setup.h"
+#include "dali/kernels/imgproc/paste/paste_gpu_input.h"
+
+namespace dali {
+namespace kernels {
+namespace paste {
+
+template<class InputType, int ndims>
+struct GridCellGPU {
+    const InputType *in;
+    ivec<ndims> cell_start, cell_end, in_anchor, in_pitch;
+};
```

Comment:
Grid is, by definition, something regular, sostart/end/anchor and pitch dont look right. Or is it more like `PatchDesc`?

Reply:
True, changed name to PatchDesc